tests/typename/typename_test.ts

```diff
@@ -112,3 +96,27 @@ Meta.test("Typename in prisma runtime", async (t) => {
       .on(e);
   });
 });
+
+Meta.test("Typename on union", async (t) => {
+  const e = await t.engine("typename/typename.py", { secrets });
+
+  await t.should("get variant type name", async () => {
+    await gql`
+      query {
+        getRgbColor {
+          color {
+            ... on RgbColor {
+              r g b __typename
```

Comment:
@Yohe-Am Is this what you meant for [the ticket](https://linear.app/metatypedev/issue/MET-642/gate-typename-on-union-selections-should-hold-member-title)?

Reply:
Yep, this looks good.